src/transformStateWithClones.js

```diff
@@ -6,8 +6,40 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  let copy = { ...state };
+
+  for (const object of actions) {
+    switch (object.type) {
+      case 'addProperties':
+        copy = Object.assign({}, copy, object.extraData);
+        break;
+
+      case 'removeProperties':
```

Comment:
Recommendation:
You can write removeProperties without creating a separate function.
```
 case 'removeProperties':
        for (const key of action.keysToRemove) {
          delete copy[key];
        }
        break;
```